realm/realm-jni/src/io_realm_internal_TableView.cpp

```diff
@@ -85,30 +85,33 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableView_nativeDistinctMulti(
     if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
         return;
     try {
+        TableView* tv = TV(nativeViewPtr);
         JniLongArray indexes(env, columnIndexes);
         jsize indexes_len = indexes.len();
-        std::vector<size_t> columns;
+        std::vector<std::vector<size_t>> columns;
+        std::vector<bool> ascending;
         for (int i = 0; i < indexes_len; ++i) {
-            if (!COL_INDEX_VALID(env, TV(nativeViewPtr), indexes[i])) {
+            if (!COL_INDEX_VALID(env, tv, indexes[i])) {
                 return;
             }
-            if (!TV(nativeViewPtr)->get_parent().has_search_index(S(indexes[i]))) {
+            if (!tv->get_parent().has_search_index(S(indexes[i]))) {
                 ThrowException(env, IllegalArgument, "The field must be indexed before distinct(...) can be used.");
                 return;
             }
-            switch (TV(nativeViewPtr)->get_column_type(S(indexes[i]))) {
+            switch (tv->get_column_type(S(indexes[i]))) {
                 case type_Bool:
                 case type_Int:
                 case type_String:
                 case type_Timestamp:
-                    columns.push_back(S(indexes[i]));
+                    columns.push_back(std::vector<size_t> { S(indexes[i]) });
+                    ascending.push_back(true);
                     break;
                 default:
                     ThrowException(env, IllegalArgument, "Invalid type - Only String, Date, boolean, byte, short, int, long and their boxed variants are supported.");
                     return;
             }
         }
-        TV(nativeViewPtr)->distinct(columns);
+        tv->distinct(SortDescriptor(tv->get_parent(), columns, ascending));
```

Comment:
The ascending parameter to `SortDescriptor` is optional, it might make sense to leave it out here when doing a distinct.